test/metabase/query_processor/middleware/fetch_source_query_test.clj

```diff
@@ -236,3 +236,28 @@
                            (str "Failed to save Card:" e)))]
         (or save-error
             (resolve-card-id-source-tables (circular-source-query card-1-id)))))))
+
+;; Alow complex dependency topologies
+(expect
+  (tt/with-temp* [Card [{card-1-id :id} {:dataset_query {:database (data/id)
+                                                         :type     :query
+                                                         :query    {:source-table (data/id :venues)}}
```

Comment:
You can/should just do `(data/mbql-query venues)` instead of typing out all the boilerplater here.

Also, `with-temp` has defaults for all the models in `metabase.test.util`. So you don't need to add stuff for anything besides `:dataset_query` here. (Sometimes in the REPL you might need to require `metabase.test.util` to get this working.)

Reply:
Ugh, thanks for noticing. Of course this is a REPL remnant